components/user-mgt/org.wso2.carbon.user.mgt/src/main/java/org/wso2/carbon/user/mgt/listeners/UserManagementAuditLogger.java

```diff
@@ -223,7 +226,15 @@ public boolean doPostUpdateCredentialByAdmin(String userName, Object newCredenti
             UserStoreManager userStoreManager) {
 
         if (isEnable()) {
-            audit.info(createAuditMessage(ListenerUtils.CHANGE_PASSWORD_BY_ADMIN_ACTION, getTargetForAuditLog
+            IdentityContext identityContext = IdentityContext.getThreadLocalIdentityContext();
+            String initiatingPersona = String.valueOf(identityContext.getFlow().getInitiatingPersona());
+            String auditMessageAction = ListenerUtils.CHANGE_PASSWORD_ACTION;
+            if (initiatingPersona.equals(USER)) {
+                auditMessageAction = ListenerUtils.CHANGE_PASSWORD_BY_USER_ACTION;
+            } else if (initiatingPersona.equals(ADMIN)) {
+                auditMessageAction = ListenerUtils.CHANGE_PASSWORD_BY_ADMIN_ACTION;
+            }
```

Comment:
Let's use switch case here as https://github.com/wso2/carbon-identity-framework/blob/712662e074f569aa2c969e2f55cdba9a87a60ed9/components/user-mgt/org.wso2.carbon.identity.user.pre.update.password.action/src/main/java/org/wso2/carbon/identity/user/pre/update/password/action/internal/execution/PreUpdatePasswordRequestBuilder.java#L119C9-L128C10

I think the APPLICATION initiatingPersona should also needs to be considered as an admin updating password.